File: Projects/PythonProject/Memory_Management/GarbageCollectionExample.py
import gc

class MyObject:
    def __init__(self, name):
        self.name = name

    def __del__(self):
        print(f"{self.name} is being collected!")

def demonstrate_garbage_collection():
    obj1 = MyObject("Object1")
    obj1 = None  # Eligible for garbage collection

    obj2 = MyObject("Object2")

    # Manually calling __del__ (not recommended in real apps, just for demo)
    obj2.__del__()

    # Suggest garbage collection
    gc.collect()

if __name__ == "__main__":
    demonstrate_garbage_collection()

File: Projects/JavaProject/src/com/Memory_Management/GarbageCollectionExample.java
package com.Memory_Management;

//In a gaming application, garbage collection ensures unused game objects 
//(e.g., expired power-ups) are removed from memory.
public class GarbageCollectionExample {

    public static void demonstrateGarbageCollection() {
        MyObject obj1 = new MyObject("Object1");
        obj1 = null; // Object1 eligible for garbage collection

        MyObject obj2 = new MyObject("Object2");
        
        obj2.finalize();

        System.gc(); // Suggest Garbage Collection
    }

    static class MyObject {
        private String name;

        public MyObject(String name) {
            this.name = name;
        }

        @Override
        protected void finalize() {
            System.out.println(name + " is being collected!");
        }
    }

    public static void main(String[] args) {
        demonstrateGarbageCollection();
    }
}